### PR TITLE
 Forwarder Encapsulation support.

### PR DESCRIPTION
**Pull request by MichalFoksa**
1. Added support of Forwarder Encapsulation. Mqtt-SN packets are encapsulated according to MQTT-SN Protocol Specification v1.2, chapter 5.5.

2. All log messages to stderr prefixed with current time, e.g:
       17:21:10 mqtt-sn-tools-8164 sending CONNECT packet...
       17:21:10 Waiting for a packet...
       17:21:10 Received  3 bytes. Type=CONNACK on Socket: 3
       17:21:10 CONNACK return code: 0x00

3. Added ignore of exe files into .gitignore.

### CODE REVIEW

File: mqtt-sn-serial-bridge.c [hunk 1]
```diff
@@ -56,42 +58,61 @@ static void usage()
...
-        default:
-            usage();
-        break;
-    }
+	while ((ch = getopt_long (argc , argv , "b:dh:p:?" , long_options , &option_index )) != -1 ) 
+	{
+
+		switch (ch) 
+		{
+			case 'b' :
```
Comment (njh) [inline mqtt-sn-serial-bridge.c]: Looks like some whitespace problems here?

File: mqtt-sn-serial-bridge.c [hunk 2]
```diff
@@ -253,9 +284,9 @@ int main(int argc, char* argv[])
     while (keep_running) {
         fd_set fdset;
 
-        FD_ZERO(&fdset);
-        FD_SET(fd, &fdset);
-        FD_SET(sock, &fdset);
+        FD_ZERO(&fdset);			// Clear the socket set
+        FD_SET(fd, &fdset);			// Add serial into fdset
+        FD_SET(sock, &fdset);		// Add socket into fdset
```
Comment (njh) [inline mqtt-sn-serial-bridge.c]: More whitespace problems?

**Comment by njh** [comment]
Would it be possible to split 1, 2 and 3 into separate PRs please?

I am not sure about the way you are generating/displaying timestamps using __CUR_TIME__.
I think it would be better to have a log() method.

For 3, it would be better to just add *.exe to .gitignore

**Comment by MichalFoksa** [comment]
Nicholas,

I do not know how to create 3 separate pull requests, I think it is even more complicated when I fix "white space problems" cause I will create additional commits with all, already committed functionalities.

**Pull request closed without merging**